lib/rules/prefer-regex-literals.js

```diff
@@ -43,6 +36,75 @@ function isStaticTemplateLiteral(node) {
     return node.type === "TemplateLiteral" && node.expressions.length === 0;
 }
 
+const validPrecedingTokens = [
+    "(",
+    ";",
+    "[",
+    ",",
+    "=",
+    "+",
+    "*",
+    "-",
+    "?",
+    "~",
+    "%",
+    "**",
+    "!",
+    "typeof",
+    "instanceof",
+    "&&",
+    "||",
+    "??",
+    "await",
+    "yield",
+    "return",
+    "...",
+    "delete",
+    "void",
+    "in",
+    "<",
+    ">",
+    "<=",
+    ">=",
+    "==",
+    "===",
+    "!=",
+    "!==",
+    "<<",
+    ">>",
+    ">>>",
+    "&",
+    "|",
+    "^",
+    ":",
+    "{",
+    "=>",
+    "*=",
+    "<<=",
+    ">>=",
+    ">>>=",
+    "^=",
+    "|=",
+    "&=",
+    "??=",
+    "||=",
+    "&&=",
+    "**=",
+    "+=",
+    "-=",
+    "/=",
+    "%=",
+    "/",
+    "do",
+    "break",
+    "continue",
+    "debugger",
+    "case",
+    "throw",
+    "of",
```

Comment:
This list can include only words that are never allowed as identifiers. Those are [reserved words](https://tc39.es/ecma262/#prod-ReservedWord) (except for `yield` and `await`, which are contextually allowed as identifiers).

`of` is not a reserved word, it can appear as identifier, and autofixing this code would produce a parsing error:

```js
x = of
RegExp("y");
```

So, `of`, `yield` and `await` shouldn't be in this list.